src/Microsoft.QualityGuidelines.Analyzers/UnitTests/MarkMembersAsStaticTests.cs

```diff
@@ -9,12 +10,359 @@ public class MarkMembersAsStaticTests : DiagnosticAnalyzerTestBase
+    }
+    
+    public int Property
+    {
+        get { return 5; }
+    }
+
+    public int Property2
+    {
+        set { s_field = value; }
+    }
+
+    public int MyProperty
+    {
+        get { return 10; }
+        set { Console.WriteLine(value); }
+    }
+
+    public event System.EventHandler<System.EventArgs> CustomEvent { add {} remove {} }
+}",
+                GetCSharpResultAt(7, 16, "Method1"),
+                GetCSharpResultAt(12, 17, "Method2"),
+                GetCSharpResultAt(14, 17, "Method3"),
+                GetCSharpResultAt(19, 16, "Method4"),
+                GetCSharpResultAt(24, 16, "Property"),
+                GetCSharpResultAt(29, 16, "Property2"),
+                GetCSharpResultAt(34, 16, "MyProperty"),
+                GetCSharpResultAt(40, 56, "CustomEvent"));
+        }
+
+        [Fact]
+        public void BasicSimpleMembers()
+        {
+            VerifyBasic(@"
+Public Class MembersTests
+    Shared s_field As Integer
+    Public Const Zero As Integer = 0
+
+    Public Function Method1(name As String) As Integer
+        Return name.Length
+    End Function
+
+    Public Sub Method2()
+    End Sub
+
+    Public Sub Method3()
+        s_field = 4
+    End Sub
+
+    Public Function Method4() As Integer
+        Return Zero
+    End Function
+
+    Public ReadOnly Property Property1 As Integer
+        Get
+            Return 5
+        End Get
+    End Property
+
+    Public WriteOnly Property Property2 As Integer
+        Set
+            s_field = Value
+        End Set
+    End Property
+
+    Public Property MyProperty As Integer
+        Get 
+            Return 10
+        End Get
+        Set
+            System.Console.WriteLine(Value)
+        End Set
+    End Property
+
+    Public Custom Event CustomEvent As EventHandler(Of EventArgs)
+        AddHandler(value As EventHandler(Of EventArgs))
+        End AddHandler
+        RemoveHandler(value As EventHandler(Of EventArgs))
+        End RemoveHandler
+        RaiseEvent(sender As Object, e As EventArgs)
+        End RaiseEvent
+    End Event
+End Class
+",
+                GetBasicResultAt(6, 21, "Method1"),
+                GetBasicResultAt(10, 16, "Method2"),
+                GetBasicResultAt(13, 16, "Method3"),
+                GetBasicResultAt(17, 21, "Method4"),
+                GetBasicResultAt(21, 30, "Property1"),
+                GetBasicResultAt(27, 31, "Property2"),
+                GetBasicResultAt(33, 21, "MyProperty"),
+                GetBasicResultAt(42, 25, "CustomEvent"));
+        }
+
+        [Fact]
+        public void CSharpSimpleMembers_NoDiagnostic()
+        {
+            VerifyCSharp(@"
+public class MembersTests
+{
+    public MembersTests() { }
+
+    public ~MembersTests() { }
+
+    public int x; 
+
+    public int Method1(string name)
+    {
+        return x;
+    }
+
+    public int Method2()
+    {
+        MembersTests temp = this;
+        return temp.x;
+    }
+
+    public int AutoProp { get; set; }
```

Comment:
I guess this AutoProp also covers the case where the backing variable is explicitly defined, but may be good to add that to ensure IOperation's `InstanceReferenceExpression` is working correctly